src/Clide.Interfaces/IMessageBoxService.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Clide
+{
+	using System.Windows;
+
+	/// <summary>
+	/// Provides a contract to show messages to the user.
+	/// </summary>
+	public interface IMessageBoxService
+	{
+		/// <summary>
+		/// Shows a message to the user.
+		/// </summary>
+		/// <returns>
+		/// <see langword="true"/> if the user clicked on Yes/OK, 
+		/// <see langword="false"/> if the user clicked No, 
+		/// <see langword="null"/> if the user cancelled the dialog or clicked 
+		/// <c>Cancel</c> or any other value other than the Yes/OK/No.
+		/// </returns>
+		bool? Show(string message, string title = MessageBoxServiceDefaults.DefaultTitle, MessageBoxButton button = MessageBoxButton.OK, MessageBoxImage icon = MessageBoxImage.None, MessageBoxResult defaultResult = MessageBoxResult.OK);
+
+		/// <summary>
+		/// Prompts the user for a response.
+		/// </summary>
+		MessageBoxResult Prompt(string message, string title = MessageBoxServiceDefaults.DefaultTitle, MessageBoxButton button = MessageBoxButton.OKCancel, MessageBoxImage icon = MessageBoxImage.Question, MessageBoxResult defaultResult = MessageBoxResult.OK);
+	}
+
+	public class MessageBoxServiceDefaults
+	{
+		public const string DefaultTitle = "Microsoft Visual Studio";
+		public const MessageBoxButton DefaultButton = MessageBoxButton.OK;
```

Comment:
I guess we should use the button default constants in the arguments above. I think that was the original intention but didn't made the change afterwards :(. 

Actually, given that both methods have different defaults, I don't think it even makes sense to have these constants for buttons here.